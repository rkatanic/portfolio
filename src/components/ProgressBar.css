.progress-bar {
  position: absolute;
  content: "";
  top: 50%;
  margin-top: 5vh;
  right: 0;
  width: 2px;
  height: 50%;
  background: linear-gradient(60deg, var(--color-body), var(--color-primary));
  border-radius: 8px;
}

.progress-bar::before {
  position: absolute;
  content: "";
  width: 8px;
  height: 8px;
  left: -3px;
  top: 0;
  z-index: 1;
  transform: rotate(45deg);
  background-color: var(--color-primary);
}

.progress-bar-pulse {
  position: absolute;
  content: "";
  width: 8px;
  height: 8px;
  left: -3px;
  top: 0;
  opacity: 0;
  border: 1px solid var(--color-primary-light);
  animation: pulsate 1s 0.35s ease-out;
  animation-iteration-count: 3;
}

.progress-bar-percent {
  transition: 0.3s;
  left: 0;
  width: 2px;
  border-radius: 8px;
  background: linear-gradient(
    60deg,
    var(--color-primary-light),
    var(--color-body)
  );
  position: absolute;
  bottom: 100%;
}

@media screen and (max-width: 767px) {
  .progress-bar {
    display: none;
  }
}
