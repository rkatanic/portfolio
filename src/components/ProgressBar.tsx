import { useState } from "react";

import "./ProgressBar.css";

const ProgressBar = (): JSX.Element => {
  const [scroll, setScroll] = useState(0);

  const onScroll = (): void => {
    const scrolled = document.documentElement.scrollTop;
    const maxHeight =
      document.documentElement.scrollHeight -
      document.documentElement.clientHeight;
    const scrollPercent = (scrolled / maxHeight) * 100;
    setScroll(scrollPercent);
  };

  window.addEventListener("scroll", onScroll);

  return (
    <div className="progress-bar">
      <div className="progress-bar-pulse"></div>
      <div
        className="progress-bar-percent"
        style={{ height: `${scroll}%` }}
      ></div>
    </div>
  );
};

export default ProgressBar;
