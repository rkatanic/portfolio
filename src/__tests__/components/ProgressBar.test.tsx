import { fireEvent, render } from "@testing-library/react";
import ProgressBar from "../../components/ProgressBar";

describe("ProgressBar", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<ProgressBar />);

    expect(baseElement).toMatchSnapshot();
  });
});
