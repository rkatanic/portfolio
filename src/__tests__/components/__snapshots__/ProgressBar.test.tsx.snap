// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render 1`] = `
<body>
  <div>
    <div
      class="progress-bar"
    >
      <div
        class="progress-bar-pulse"
      />
      <div
        class="progress-bar-percent"
        style="height: 0%;"
      />
    </div>
  </div>
</body>
`;
